multigeiger/speaker.cpp

```diff
@@ -31,34 +52,43 @@ void cycle(int t1, int t2, int volume) {
   delayMicroseconds(t2);
 }
 
-void tick(int use_led, int use_speaker) {
-  // make LED flicker and speaker tick
-  if (use_led) {
-    digitalWrite(LED_BUILTIN, HIGH);    // switch on LED
-  }
-  for (int t = 0; t < 4; t++) {
-    if (use_speaker)
-      cycle(500, 500, 1);  // takes 1ms
-    else if (use_led)
-      delay(1);            // also takes 1ms
-  }
-  if (use_led) {
-    digitalWrite(LED_BUILTIN, LOW);     // switch off LED
-  }
-}
-
-void tone(int frequency_mHz, int time_ms, int volume) {
+void tone(int frequency_mHz, int duration_ms, int volume) {
+  // volume: 0: low, 1: high
+  #if PWM
+  mcpwm_set_frequency(MCPWM_UNIT_0, MCPWM_TIMER_0, frequency_mHz / 1000);
+  // enable sound output
+  float duty;
+  duty = volume ? 50.0 : 10.0;
```

Comment:
maybe not the best way to do this, last tone of melody sounds a bit different than previously.

Reply:
@rexfue this is why the startup sound sounds a bit ugly at the end, because the last note has a duty cycle of 10% (not 50% as all others).

I guess I'll try changing it to 50% also and hear if it gets better then.

Reply:
at 50% it sounds ok.